dist/js/components/AmountWidget.js

```diff
@@ -4,17 +4,9 @@ import BaseWidget from "./BaseWidget.js";
 class AmountWidget extends BaseWidget {
   constructor(element) {
     super(element, settings.amountWidget.defaultValue);
-
-    const thisWidget = this;
-
-    // console.log('AmountWidget:', thisWidget);
-    // console.log('constructor arguments:', element);
-
-    thisWidget.getElements(element);
-    thisWidget.setValue(
-      thisWidget.input.value || settings.amountWidget.defaultValue
-    );
-    thisWidget.initActions();
+    this.getElements(element);
+    this.initActions();
+    this.value = settings.amountWidget.defaultValue;
```

Comment:
W ten sposób sprawiasz, że AmountWidget zawsze po inicjalizacji ma mieć ustawioną wartość domyślną. To z kolei rodzi bugi, bo w niektórych sytuacjach takiego zachowania nie chcemy np. gdy dodasz do koszyka 5 jednakowych produktów na raz, chcesz, aby AmountWidget w koszyku pokazywał 5, a nie wartość domyślną - 1.
Powinnaś to zrobić wywołując metodę `setValue` widgetu, tak jak to robiłaś wcześniej. Jako parametr powinna przyjąć wartość inputa lub wartość domyślną, czyli tak jak było wcześniej.
Problemem było natomiast to, że źle wskazywałaś, gdzie WidgetAmount przechowuje referencję do inputa. Nie jest to `thisWidget.input`. Możesz, albo wylogować sobie zawartość `thisWidget` i sprawdzić, gdzie dokładnie przechowywana jest referencja do inputa, albo sprawdzić to w metodzie `getElements`, która zapisuje wszystkie referencje do węzłów DOM